logger/logger.go

```diff
@@ -9,7 +9,15 @@ import (
 )
 
 type TfLogger struct {
-	Name string
+	ctx context.Context
+}
+
+func NewTfLogger(ctx context.Context) *TfLogger {
```

Comment:
[nit] I would just add a small comment here to explain that this function is expecting a certain type of context that has contains a logger and not any context.

Reply:
Added